packages/react-devtools-shared/src/devtools/views/Profiler/ProfilerContext.js

```diff
@@ -235,6 +247,29 @@ function ProfilerContextController({children}: Props) {
     });
   }
 
+  const [numFilteredCommits, setNumFilteredCommits] = useState<number>(0);
+  const [
+    selectedFilteredCommitIndex,
+    setSelectedFilteredCommitIndex,
+  ] = useState<number | null>(null);
+
+  // TODO (ProfilerContext) This should be managed by the context controller (reducer).
+  useEffect(() => {
+    if (selectedFilteredCommitIndex === null) {
+      if (numFilteredCommits > 0) {
+        selectCommitIndex(0);
+      } else {
+        selectCommitIndex(null);
+      }
+    } else if (selectedFilteredCommitIndex >= numFilteredCommits) {
+      if (numFilteredCommits === 0) {
+        selectCommitIndex(null);
+      } else {
+        selectCommitIndex(numFilteredCommits - 1);
+      }
+    }
+  }, [selectedFilteredCommitIndex, numFilteredCommits]);
```

Comment:
This is probably a wrong way to do it in ProfilerContext, how to implement it better? Maybe use EventEmitter in SnapshotSelector which will fire update events? Or create a reducer hook which will manage selectCommitIndex calls? Should be this reducer used instead of selectCommitIndex calls in all the places?

Reply:
Just found related [issue](https://github.com/facebook/react/issues/16444), which state variables should manage that reducer?